### DISCUSSION TITLE
Is there a good way to get path parameters?

### DISCUSSION THREAD

**Discussion by jellybobbin** [tokio-rs/axum#1459] [Q&A] [answered]
### route in actix_web:
`xxx.com/{user_id}-aa/{post_id}bb.html`

### when request url:
```
xxx.com/11-aa/22bb.html

//get parameters:
user_id: 11, post_id:22
```

**Comment by davidpdrsn** [comment]
https://docs.rs/axum/0.6.0-rc.2/axum/struct.Router.html#captures

**Reply by jellybobbin** [reply]
I know that in Axum: 

`xxx.com/:user_id/:post_id`

can be get `user_id` and `post_id`

But `:user_id` and `{user_id}-aa` are different。

The same parameters are in `axum`:

`xxx.com/:user_id-aa/:post_idbb.html`

The meaning is completely different...

**Reply by davidpdrsn** [reply]
You have to use `:user_id` and remove `-aa` yourself in the handler.

**Reply by jellybobbin** [reply]
But I hope my url is like this:

`xxx.com/{user_id}-aa/{post_id}bb.html`

Is there a better way？

**Reply by jellybobbin** [reply]
The problem is that axum use `:variable` as the parameter parse；

This leads to the inability to do the same thing as `actix-web`

in `xxx.com/:user_id-foo/:post_idbar.html`:

variable is `user_id` instead of `user_id-foo`

Variable is `post_id` instead of `post_idbar.html`

**Reply by davidpdrsn** [reply, marked as answer]
Nope. You have to do `xxx.com/:user_id/:post_id` and parse things yourself in the handler.

**Reply by jellybobbin** [reply]
Ok, thank you very much...
